site/source/docs/porting/pthreads.rst

```diff
@@ -43,17 +43,82 @@ but is unrelated. That flag does not use pthreads or SharedArrayBuffer, and
 instead uses a plain Web Worker to run your main program (and postMessage to
 proxy messages back and forth).
 
+Proxying
+========
+
+The Web allows certain operations to only happen from the main browser thread,
+like interacting with the DOM. As a result, various operations are proxied to
+the main browser thread if they are called on a background thread. See
+`bug 3495 <https://github.com/emscripten-core/emscripten/issues/3495>`_ for
+more information and how to try to work around this until this. To check which
+operations are proxied, you can look for the function's implementation in
+the JS library (``src/library_*``) and see if it is annotated with
+``__proxy: 'sync'`` or ``__proxy: 'async'``; however, note that the browser
+itself proxies certain things (like some GL operations), so there is no
+general way to be safe here (aside from not blocking on the main browser
+thread).
+
+In addition, Emscripten currently has a simple model of file I/O only happening
+on the main application thread (as we support JS plugin filesystems, which
+cannot share memory), which is another set of operations that are proxied.
```

Comment:
I think this would be clearer if "), which is another set" -> "); this is another". Assuming I'm correct in my interpretation that this means that all file operations may only happen on the main thread, so they are proxied.